OutOfSchool/OutOfSchool.WebApi/Services/Database/WorkshopService.cs

```diff
@@ -266,22 +269,100 @@ public async Task<SearchResult<T>> GetByProviderId<T>(Guid id, ExcludeIdFilter f
             filter.ExcludedId == null
                 ? (x.ProviderId == id)
                 : (x.ProviderId == id && x.Id != filter.ExcludedId)).ConfigureAwait(false);
-        var workshops = await workshopRepository.Get(
+
+        var workshops = workshopRepository.Get(
                 skip: filter.From,
                 take: filter.Size,
                 includeProperties: includingPropertiesForMappingDtoModel,
                 where: x => filter.ExcludedId == null
                     ? (x.ProviderId == id)
-                    : (x.ProviderId == id && x.Id != filter.ExcludedId))
-            .ToListAsync()
-            .ConfigureAwait(false);
-
-        logger.LogInformation(!workshops.Any()
+                    : (x.ProviderId == id && x.Id != filter.ExcludedId));
+
+        var chatrooms = roomRepository.Get(
+            includeProperties: $"{nameof(ChatRoomWorkshop.ChatMessages)}",
+            where: x => x.ChatMessages.Where(x => x.ReadDateTime == null && !x.SenderRoleIsProvider).Any());
+
+        var query = from w in workshops
```

Comment:
(Code style) This method demonstrates both LINQ syntax/style - chained (see `workshopsWithUnreadMessages.Select` below) and query one. 
General recommendation is to use only one, my own recommendation (taking code maintainability into account) is to prefer chained.
Please redesign your code.